components/eam/src/physics/rrtmg/radiation.F90

```diff
@@ -1408,6 +1408,7 @@ subroutine radiation_tend(state,ptend, pbuf, &
                   call outfld('SWCF'//diag(icall),swcf  ,pcols,lchnk)
 
 #if defined(CLDERA_PROFILING)
+                  call cldera_switch_context("eam")
```

Comment:
does this need to be added whenever we set ctools fields? I think we do it in a lot more files.

Reply:
It shouldn't. I think we do call this early in the eam run stack, somewhere in cam_comp.F90 maybe.

Reply:
yes, then maybe we can remove this one? or was it required for some reason?

Reply:
This one caused many weeks of inscrutable errors with the multiple profiling contexts enabled. It is definitely needed and will cause crashes otherwise. This has to do with the half-time stepping coupling in the atm module I believe.

Reply:
I'm just wondering if maybe we should change the context in a single place higher up the chain to avoid future errors if hooks for some other physics model gets added (i.e. cirrus stuff?).

Reply:
Ah, in `atm_comp_mct.F90` I see the switch context call inside run and final, but not init. That may be the pb. Atm's init is called twice in E3SM (long story): the 1st call is before lnd, while the 2nd is after lnd. The 1st call is OK even without a context switch, since the new context is created. During the second call, while we do not init anything, EAM does call some cam_runX routines (due to how the atm step is misaligned with the cpl step), mostly to compute compatible surf fluxes for the surf model(s). This is where things get messy, since at this point, when running with both lnd/atm, the context is still stuck on lnd.

Reply:
I suggest adding the switch context call in atm_comp_mct.F90, around line 400:
```
    else
    
       ! For initial run, run cam radiation/clouds and return
       ! For restart run, read restart x2a_a
       ! Note - a2x_a is computed upon the completion of the previous run - cam_run1 is called
       ! only for the purposes of finishing the flux averaged calculation to compute a2x_a
       ! Note - cam_run1 is called on restart only to have cam internal state consistent with the 
       ! a2x_a state sent to the coupler
       
       < ADD SWITCH CONTEXT HERE>
```

Reply:
Yep, that looks like it works, so I'll go with it. Thanks!